ambry-protocol/src/test/java/com.github.ambry.protocol/RequestResponseTest.java

```diff
@@ -293,28 +292,46 @@ public void getRequestResponseTest() throws IOException {
   public void deleteRequestResponseTest() throws IOException {
     MockClusterMap clusterMap = new MockClusterMap();
     BlobId id1 = new BlobId(clusterMap.getWritablePartitionIds().get(0));
-    DeleteRequest deleteRequest = new DeleteRequest(1234, "client", id1);
-    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-    WritableByteChannel writableByteChannel = Channels.newChannel(outputStream);
-    do {
-      deleteRequest.writeTo(writableByteChannel);
-    } while (!deleteRequest.isSendComplete());
-    DataInputStream requestStream = new DataInputStream(new ByteArrayInputStream(outputStream.toByteArray()));
-    requestStream.readLong(); // read length
-    requestStream.readShort(); // read short
-    DeleteRequest deserializedDeleteRequest = DeleteRequest.readFrom(requestStream, clusterMap);
-    Assert.assertEquals(deserializedDeleteRequest.getClientId(), "client");
-    Assert.assertEquals(deserializedDeleteRequest.getBlobId(), id1);
-    DeleteResponse response = new DeleteResponse(1234, "client", ServerErrorCode.No_Error);
-    outputStream.reset();
-    do {
-      response.writeTo(writableByteChannel);
-    } while (!response.isSendComplete());
-    requestStream = new DataInputStream(new ByteArrayInputStream(outputStream.toByteArray()));
-    requestStream.readLong(); // read size
-    DeleteResponse deserializedDeleteResponse = DeleteResponse.readFrom(requestStream);
-    Assert.assertEquals(deserializedDeleteResponse.getCorrelationId(), 1234);
-    Assert.assertEquals(deserializedDeleteResponse.getError(), ServerErrorCode.No_Error);
+    short[] versions = new short[]{DeleteRequest.Delete_Request_Version_1, DeleteRequest.Delete_Request_Version_2};
+    for (short version : versions) {
+      short accountId = Utils.getRandomShort(random);
+      short containerId = Utils.getRandomShort(random);
+      int deletionTimeMs =
```

Comment:
why is this cast to `int`? `deletionTimeMs` can be `long` right? This is probably in overflow